contracts/contracts/vault/OETHVaultCore.sol

```diff
@@ -143,6 +155,308 @@ contract OETHVaultCore is VaultCore {
+     * @param _requestId Unique ID for the withdrawal request
+     * @return amount Amount of WETH transferred to the withdrawer
+     */
+    function claimWithdrawal(uint256 _requestId)
+        external
+        whenNotCapitalPaused
+        nonReentrant
+        returns (uint256 amount)
+    {
+        amount = _claimWithdrawal(_requestId);
+
+        // transfer WETH from the vault to the withdrawer
+        IERC20(weth).safeTransfer(msg.sender, amount);
+
+        // Prevent insolvency
+        _postRedeem(amount);
+    }
+
+    /**
+     * @notice Claim a previously requested withdrawals once they are claimable.
+     * This requests can be claimed once the withdrawal queue's `claimable` amount
+     * is greater than or equal each request's `queued` amount and 30 minutes has passed.
+     * If one of the requests is not claimable, the whole transaction will revert with `Queue pending liquidity`.
+     * If one of the requests is not older than 30 minutes,
+     * the whole transaction will revert with `Claim delay not met`.
+     * @param _requestIds Unique ID of each withdrawal request
+     * @return amounts Amount of WETH received for each request
+     * @return totalAmount Total amount of WETH transferred to the withdrawer
+     */
+    function claimWithdrawals(uint256[] memory _requestIds)
+        external
+        whenNotCapitalPaused
+        nonReentrant
+        returns (uint256[] memory amounts, uint256 totalAmount)
+    {
+        amounts = new uint256[](_requestIds.length);
+        for (uint256 i = 0; i < _requestIds.length; ++i) {
+            amounts[i] = _claimWithdrawal(_requestIds[i]);
+            totalAmount += amounts[i];
+        }
+
+        // transfer all the claimed WETH from the vault to the withdrawer
+        IERC20(weth).safeTransfer(msg.sender, totalAmount);
+
+        // Prevent insolvency
+        _postRedeem(totalAmount);
+    }
+
+    // slither-disable-start reentrancy-no-eth
+
+    function _claimWithdrawal(uint256 requestId)
+        internal
+        returns (uint256 amount)
+    {
+        // Check if there's enough liquidity to cover the withdrawal request
+        WithdrawalQueueMetadata memory queue = withdrawalQueueMetadata;
+        WithdrawalRequest memory request = withdrawalRequests[requestId];
+
+        require(request.claimed == false, "Already claimed");
+        require(request.withdrawer == msg.sender, "Not requester");
+        require(
+            request.timestamp + CLAIM_DELAY <= block.timestamp,
+            "Claim delay not met"
+        );
+
+        // Try and get more liquidity in the withdrawal queue if there is not enough
+        if (request.queued > queue.claimable) {
+            // Stream any harvested rewards (WETH) that are available to the Vault
+            IDripper(dripper).collect();
+
+            // Add any WETH from the Dripper to the withdrawal queue
+            uint256 addedClaimable = _addWithdrawalQueueLiquidity();
```

Comment:
🟧 Strong personal preference here against this code.

We have a dangerous pattern here in this method, where we check that the request is claimed, then make an external state changing call, and then set that the request has been claimed.

This is a classic setup for a reentrancy vulnerability. If the external call would allow an attacker to call back in, we could claim over and over again. This is why we had to add slither exception to this method. (Yes, they would have to setup the accounting just right to make this work) (yes, dripper should be safe)

1) As a general rule of thumb, whenever we check some kind of one-time only gate value, like `request.claimed`, we should immediately set it right in the next line of code. In this way it's not vulnerable to reentrancy issues, or missing getting set because some if statement returned somewhere.

For example, from primeETH, this is a good way to handle this:

<img width="447" alt="image" src="https://github.com/user-attachments/assets/d1ad2196-4daf-4fd9-a939-3ec7f1fa69e6">

2) It's awkward having an external call and global accounting update right in the middle of all this internal request queue updating. Also makes the math ugly. Can we do this after loading the structs from storage, before we do the require checks and then updates on the request? If it needs to run, we can reload the queue struct after. 

3) I have a small preference for not having a multi `claimWithdrawals`. It's one more code path that could go wrong. I'm open though if it's useful.

Reply:
1. This makes sense. I've marked the request at claimed immediately after it's checked. I also changed the order of the requires as `Claim delay not met` is the more likely revert reason.

Reply:
2. I've done a number of interactions refactoring the claim functions. 

`claimWithdrawal` will only call the Dripper and do the global accounting if there is not enough liquidity to claim. These are now done before calling `_claimWithdrawal`

`_claimWithdrawals` doesn't try to work out what the last request is in the parameter list. It calls the Dripper and does the global accounting regardless.

`_claimWithdrawal` is an internal function with no external calls

Reply:
3. The frontend is using  `claimWithdrawal` and `claimWithdrawals` so I'll leave them

Reply:
I like these changes.